### PR TITLE
#24358: Support vm_template for azurerm_virtual_desktop_host_pool resource

### LINKED ISSUE #24358: Support for `vmTemplate` in `azurerm_virtual_desktop_host_pool`
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

### Description

There doesn't appear to be support for `vmTemplate` property in a `azurerm_virtual_desktop_host_pool`

Allows the host pool to create VMs dynamically during ramp up.

Some Details...

https://learn.microsoft.com/en-us/rest/api/desktopvirtualization/host-pools/get?view=rest-desktopvirtualization-2022-02-10-preview&tabs=HTTP

image

### New or Affected Resource(s)/Data Source(s)

azurerm_virtual_desktop_host_pool

### Potential Terraform Configuration

_No response_

### References

_No response_

### PR DESCRIPTION
**Pull request by harshavmb**
Support `vm_template` for `azurerm_virtual_desktop_host_pool` resource. 

Fixes #24358.

### CODE REVIEW

File: internal/services/desktopvirtualization/virtual_desktop_host_pool_resource.go [hunk 1]
```diff
@@ -220,6 +227,15 @@ func resourceVirtualDesktopHostPoolCreate(d *pluginsdk.ResourceData, meta interf
 	}
 
 	personalDesktopAssignmentType := hostpool.PersonalDesktopAssignmentType(d.Get("personal_desktop_assignment_type").(string))
+	vmTemplate := utils.String(d.Get("vm_template").(string))
+	if *vmTemplate != "" {
+		// we have no use with the json object as azure accepts string only
+		// merely here for validation
+		_, err := pluginsdk.ExpandJsonFromString(*vmTemplate)
```
Comment (catriona-m) [inline internal/services/desktopvirtualization/virtual_desktop_host_pool_resource.go:234]:
```suggestion
	vmTemplate := d.Get("vm_template").(string)
	if vmTemplate != "" {
		// we have no use with the json object as azure accepts string only
		// merely here for validation
		_, err := pluginsdk.ExpandJsonFromString(vmTemplate)
```

File: website/docs/r/virtual_desktop_host_pool.html.markdown [hunk 2]
```diff
@@ -79,6 +79,8 @@ The following arguments are supported:
 
 * `scheduled_agent_updates` - (Optional) A `scheduled_agent_updates` block as defined below. This enables control of when Agent Updates will be applied to Session Hosts.
 
+* `vm_template` - (Optional) A VM template for sessionhosts configuration within hostpool. This is a JSON string.
```
Comment (catriona-m) [inline website/docs/r/virtual_desktop_host_pool.html.markdown:82]:
```suggestion
* `vm_template` - (Optional) A VM template for session hosts configuration within hostpool. This is a JSON string.
```

**Review by catriona-m — CHANGES_REQUESTED** [review]
Thanks for this @harshavmb, I've left a couple of minor comments but once those are fixed up I can take another look. It would also be good to include this property in a test. Thanks!

**Comment by harshavmb** [comment]
Hi @catriona-m ,

Indeed I wanted to add a test case but couldn't find some documentation about `vm_template`.  Also my organization has restricted has restricted access to create AVDs. 

I shall check with the request initiator if he could provide me a sample template to play with.

**Comment by catriona-m** [comment]
HI @harshavmb - just checking in to see if you were able to get the config needed to create a test for this? Thanks!

**Comment by harshavmb** [comment]
> HI @harshavmb - just checking in to see if you were able to get the config needed to create a test for this? Thanks!

Hi @catriona-m ,

I have added the test case now. Could you please re-review?

**Review by katbyte — APPROVED** [review]
Thanks @harshavmb - LGTM 🖥️

**Pull request merged**